Sources/ComposableArchitecture/SharedState/Shared.swift

```diff
@@ -408,11 +411,10 @@ extension Shared: MutableCollection
 where Value: MutableCollection & RandomAccessCollection, Value.Index: Hashable {
   public subscript(position: Value.Index) -> Shared<Value.Element> {
     get {
-      assertionFailure("Conformance of 'Shared<Value>' to 'MutableCollection' is unavailable.")
-      return self[position, default: DefaultSubscript(self.wrappedValue[position])]
+      fatalError("Conformance of 'Shared<Value>' to 'MutableCollection' is unavailable.")
     }
     set {
-      self._wrappedValue[position] = newValue.wrappedValue
+      fatalError("Conformance of 'Shared<Value>' to 'MutableCollection' is unavailable.")
```

Comment:
This subscript is unavailable so it should be ok to `fatalError`. If we don't fatal error, then we get sendability warnings that can't be fixed because you can't conditionally conform to a protocol with a constraint against a marker protocol, like `Sendable`.